test/language/class/class-name-binding.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class name binding
+---*/
+var C2;
+class C {
```

Comment:
It seems like this test is only interesting if the class is defined with a ClassExpression instead of a ClassDeclaration. As it stands, it's just asserting that class methods have access to the outer environment record.

Instead, what about

```js
var C;
var OuterName = class InnerName {
  constructor() {
    C = InnerName
  }
// etc...
```